Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -773,18 +773,14 @@ extension NavigationViewController: NavigationServiceDelegate {
             component.navigationService(service, didUpdate: progress, with: location, rawLocation: rawLocation)
         }
 
-        // If the user has arrived, don't snap the user puck.
-        // In case if user drives beyond the waypoint, we should accurately depict this.
-        guard let destination = progress.currentLeg.destination else {
-            preconditionFailure("Current leg has no destination")
-        }
-        let preventRerouting = navigationService.delegate?.navigationService(navigationService, shouldPreventReroutesWhenArrivingAt: destination) ?? RouteController.DefaultBehavior.shouldPreventReroutesWhenArrivingAtWaypoint
+        
+        let isNotRerouting = navigationService.router.userIsOnRoute(location)
```

Comment:
The previous logic did more than check whether the user is on the route. Will the puck now stop short of the end of the route line just because the navigator considers the user to be within the arrival radius?

Reply:
Since we are not controlling `shouldPreventReroutesWhenArrivingAt` check anymore, we are relying to NN implementation.

Reply:
Restored original logic